File: 2022/day1/main.py
with open("input.txt") as f:
    data = f.read()

cals = [sum(int(c) for c in cal) for cal in [s.split() for s in data.split("\n\n")]]
print(f"Part 1: {max(cals)}")
cals.sort(reverse=True)
print(f"Part 2: {sum(cals[0:3])}")

File: 2022/day1/main.go
package main

import (
	"fmt"
	"os"
	"sort"
	"strconv"
	"strings"
)


func main() {
	b,_ := os.ReadFile("input.txt")
	data := strings.Split(string(b),"\n\n")
	n := len(data)
	sums := make([]int,n)
	for i, d := range data {
		cals := strings.Split(strings.TrimSpace(d),"\n")
		s := 0
		for _, cal := range cals {
			num, _ := strconv.Atoi(cal)
			s+= num
		}
		sums[i] =  s
	}
	sort.Ints(sums)
	fmt.Println("Part 1:", sums[n-1])
	fmt.Println("Part 2:", sums[n-1] + sums[n-2] + sums[n-3])
}
